packaging/greenboot/microshift-running-check.sh

```diff
@@ -28,7 +28,43 @@ function return_failure() {
 # args: None
 # return: None
 function script_exit() {
-    [ "$?" -ne 0 ] && echo "FAILURE" || echo "FINISHED"
+    if [ "$?" -ne 0 ] ; then
+        # Always log the last 1000 MicroShift and etcd journal entries
+        for srv in microshift.service microshift-etcd.service ; do
+            log_failure_cmd "${srv}" "journalctl -xu ${srv} -n 1000 --no-pager"
+        done
+        # Always log the list of pods
```

Comment:
Could we log this sort of information closer to the place where it is being checked? For example, `"Expected N pods in $namespace, found M: $(oc get pods -n $namespace)"`

Reply:
The problem is that it's difficult because the checks are running in background and there's a wait for status reports.
I think, however, that producing pod list and events before exit helps to identify the problem on the spot.

So, there will be a message that pod restart check failed and pod-list-snapshot taken 1s after that error.

Reply:
How is a user supposed to know what pods should be present so they can compare that to the list of pods that are present?

Reply:
In the loops where the script looks for pods in specific namespaces, it knows when it does not find them. It should report whatever details it can at that point. Messages like "Expected 5 ready pods in openshift-foo but found 4" point directly to issues within the namespace. "Deployment `embedded-component` has no ready pods" is even better because it points directly to the thing that is broken.

Reply:
I added error messages in all the loops. Is this any better?